### ISSUE TITLE
Is the layering of focushighlight and nodes problematic?

### ISSUE THREAD

**Issue by zepumph** [phetsims/faradays-law#87]
Since you can move the magnet everywhere in FL, you can get to this view when highlighting some of the controls. I don't see this as a high priority problem, but I can imagine that this is going to be a problem where ever there are freely movable objects. @jessegreenberg @emily-phet is this a problem?

![image](https://user-images.githubusercontent.com/6856943/32517483-100071b4-c3b3-11e7-8de7-711cd27578d7.png)

**Comment by jessegreenberg** [comment]
Good question @zepumph, that looks weird. This is a general problem for draggable things, here it is in charges and fields:
![capture](https://user-images.githubusercontent.com/6396244/32518715-ae9cfdf6-c3c7-11e7-8627-118359cf0b95.PNG)

Focus is on a check box under the meter.

**Comment by jessegreenberg** [comment]
Maybe this is OK? It  would also be bad if the focus highlight were totally under the meter in that case. I don't know what the best approach should be. Also wondering if @terracoda has any thoughts.

**Comment by terracoda** [comment]
@jessegreenberg and @zepumph, those are interesting examples, especially the checkbox behind the meter. Though I think the magnet example looks more odd than the meter example. 

In normal focus, the focus is part of the actual object that has focus. In the case of a PhET sim object, it is like focus and the object are 2 different things.

The general idea behind focus, is that it should always be visible, see [Section 4.3 Focus VS Selection and the Perception of Dual Focus](https://www.w3.org/TR/wai-aria-practices-1.1/#kbd_focus_vs_selection).

It might be wise to not worry about this issue until we address mobile access. If PDOM elements end up being directly on top of the visual sim, then placing focus on them may present new or similar layering issues.

**Comment by zepumph** [comment]
I'm inclined to agree with @terracoda. This is a bit weird, but based on your spec. We shouldn't work on it until there is an actual problem.

**Comment by emily-phet** [comment]
Just to make sure I understand, the issue is if an object has focus, and then you use the mouse (or something else) to drag an object around? I guess I'm a little unclear about how you can be dragging an object while another object has focus.

**Comment by terracoda** [comment]
In the meter example above, the meter was was dragged in some fashion (with keyboard or mouse) to a place where it obscures the checkbox control. Then later in a separate interaction the user is tabbing through and gets keyboard focus on the checkbox control. The keyboard focus highlight is visually drawn on the top layer, so it is visible, even though the control it is highlighting is not visible.

Regarding focus highlights with mouse interactions, in general the visual cuing for mouse interactions happen on mouse over (hover), and often when pressing or activating a control (active). These visual cues can be the same or can be different. In PhET sims the two are distinctly different.

There is a style that represents an active mouse click or press on most if not all PhET button-like controls. It is worth considering if that "pressed" look should be when a user presses with a key command. I think we may have discussed this before. The pink focus highlight would remain, and the darker pressed style could be added on press, like it is added on click.

**Comment by terracoda** [comment]
@emily-phet, we can start a a11y-research topic on this if you would like to consider visual active state alignment for keyboard interactions.
